File: src/leetcode/python/easy/MinimumDepthOfBinaryTree_111.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class MinimumDepthOfBinaryTree_111(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        d = map(self.minDepth, (root.left, root.right))
        return 1 + (min(d) or max(d))

File: src/leetcode/java/easy/MinimumDepthOfBinaryTree_111.java
package leetcode.java.easy;

public class MinimumDepthOfBinaryTree_111 {
  public int minDepth(TreeNode root) {
    if (root == null) return 0;
    int l = minDepth(root.left), r = minDepth(root.right);
    return 1 + (Math.min(l, r) > 0 ? Math.min(l, r) : Math.max(l, r));
  }

  class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
  }
}